stytch/src/main/kotlin/com/stytch/java/common/PolicyCache.kt

```diff
@@ -15,25 +18,52 @@ public class PermissionException(authorizationCheck: AuthorizationCheck) :
 
 internal class PolicyCache(
     private val client: RBAC,
+    private val coroutineScope: CoroutineScope,
 ) {
     private var cachedPolicy: Policy? = null
     private var policyLastUpdate: Instant? = null
+    private var backgroundRefreshStarted = false
+
+    companion object {
+        private const val CACHE_TTL_SECONDS = 3600L // 1 hour
+        private const val REFRESH_INTERVAL_MS = 3600000L // 1 hour in milliseconds
+    }
 
     private fun getPolicy(invalidate: Boolean = false): Policy {
         val isMissing = cachedPolicy == null || policyLastUpdate == null
-        val isStale = policyLastUpdate == null || Duration.between(policyLastUpdate, Instant.now()).seconds > 300
+        val isStale = policyLastUpdate == null || Duration.between(policyLastUpdate, Instant.now()).seconds > CACHE_TTL_SECONDS
         if (invalidate || isMissing || isStale) {
-            when (val result = client.policyCompletable(PolicyRequest()).get()) {
-                is StytchResult.Success -> {
-                    cachedPolicy = result.value.policy
-                    policyLastUpdate = Instant.now()
-                }
-                else -> {}
-            }
+            refreshPolicy()
         }
+
+        // Start background refresh after first successful fetch
+        if (!backgroundRefreshStarted && cachedPolicy != null) {
+            startBackgroundRefresh()
+            backgroundRefreshStarted = true
+        }
+
         return cachedPolicy ?: throw Exception("Error fetching the policy")
     }
 
+    private fun refreshPolicy() {
+        when (val result = client.policyCompletable(PolicyRequest()).get()) {
+            is StytchResult.Success -> {
+                cachedPolicy = result.value.policy
+                policyLastUpdate = Instant.now()
+            }
+            else -> {}
+        }
+    }
+
+    private fun startBackgroundRefresh() {
+        coroutineScope.launch {
+            while (true) {
```

Comment:
instead of `while (true)`, check `while (isActive)`, to respect if the coroutine scope has been cancelled for some reason